### ISSUE TITLE
Zlib::GzipReader missing each_char iterator

### ISSUE THREAD

**Issue by headius** [jruby/jruby#8351]
As reported in https://github.com/jruby/jruby/issues/6613#issuecomment-2357326234

**Comment by headius** [comment]
@KostyaSha I don't think it would be too bad! For zlib specifically, we use a Java library called jzlib (https://github.com/jruby/jzlib) that mimics libzlib pretty closely, so the JRuby code around it can usually be a "soft" port of the CRuby logic. If I were to tackle it now, I'd look at CRuby's extension (https://github.com/ruby/zlib), find their implementation of `each_char`, and translate it to Java.

In rare cases, something is missing from jzlib and we have to add it, but I think this method would be defined in terms of other read operations.

Are you interested in attempting to do the work? I'd be happy to support you!

**Comment by KostyaSha** [comment]
sorry i had no time to implement this(

**Comment by headius** [comment]
@KostyaSha No problem, there will be other opportunities. Thanks go to @danini-the-panini for getting it done!
